config/user.js

```diff
@@ -0,0 +1,16 @@
+// -*- mode: gnome-shell -*-
+var Extension = imports.misc.extensionUtils.getCurrentExtension();
+var Tiling = Extension.imports.tiling;
+var Keybindings = Extension.imports.keybindings;
+
+function init() {
+    // Runs _only_ once on Gnome login (or Gnome shell restart)
+}
```

Comment:
@Lythenas - yeah, I'm starting to think this too.  Even more, [the latest review by extensions.gnome.org](https://extensions.gnome.org/review/43111) want use to remove that init() entirely  - so `user.js` init wouldn't be called.

Now, if they have code in user.js init(), it won't break PaperWM but it will give an error notification to the user.  I'm not sure how many use that, but it would require moving to enable/disable (which is better here anyways).

We could remove the above init(), and maybe use the template with some comments re that.

Reply:
Unfortunately I also can't tell you how many people even have code in `user.js`.

Maybe to (kind of) retain the old behavior, we could call the init of `user.js` on the very first call of `enable` (right before `enable` of the `user.js`). I expect this wouldn't break anything for the user.

Also I think it would also be reasonable for use to just remove the support for `user.js` entirely. It is not actually that much less code than creating a second extension (basically just the `metadata.json` is missing). In that second extension we (or users) would not be restricted by gnome guidelines. (We can of course still provide a template for that and/or a script to create that second extension)

Reply:
So, I pushed some changes to:

- remove `init()` from extensions.js;
- run user.js `init` on first PaperWM enable;
- also only run `warnAboutGnomeShellVersionCompatibility` on first enable.